### ISSUE TITLE
Config file of `sea-orm-cli generate entity` command

### ISSUE THREAD

**Issue by billy1624** [SeaQL/sea-orm#1185]
## Motivation

The options and flags for `sea-orm-cli generate entity` command is getting more and more. It's time to consider a scalable plan: loading the options from a config file.

## Proposed Solutions

Load all options by providing the path to the config file: `sea-orm-cli generate entity --config generate-entity.config.json`. The config file must be a JSON value.

```json
{
    "compact_format": true,
    "expanded_format": true,
    "include_hidden_tables": true,
    "tables": "my_table",
    "ignore_tables": ["seaql_migrations"],
    "max_connections": 1,
    "output_dir": "out",
    "database_schema": "public",
    "database_url": "postgres://root:root@localhost/database",
    "with_serde": "none",
    "with_copy_enums": true,
    "date_time_crate": "chrono",
    "lib": true
}
```

## Related Discussion

- https://github.com/SeaQL/sea-orm/issues/1120#issuecomment-1279875816

**Comment by billy1624** [comment]
Comments? @tyt2y3

**Comment by tyt2y3** [comment]
Agree. Although I think there should be a default path where we load this file from.
I am think we'd load it from the specified destination directory.
This would encourage projects to actually commit this file into their project.

Second, only options related to entity generation can be specified.

and the follow four
```json
    "max_connections": 1,
    "output_dir": "out",
    "database_schema": "public",
    "database_url": "postgres://root:root@localhost/database",
```

should NOT be included.

**Comment by aadi58002** [comment]
I would like to work on this issue. I will like to use the `https://github.com/mehcode/config-rs` lib. Which seem to support multiple types as multiple config file hierarchy.

**Comment by tyt2y3** [comment]
Apparently config-rs is written by one of the SQLx authors

**Comment by aadi58002** [comment]
Nice fact

**Comment by aadi58002** [comment]
Acc. To you which approach is better
1. Let the config options be defined on per argument bases and let the function each handle there config files on there own.
Advantages :-
- The option which can be placed in config and the arguments can be different and will be more flexible in the long run.
Disadvantages :-
- It requires more work (Somewhat) to implement any new function as then you have to consider that the config option will also need to be handled
- Different structs need to be defined for the config and arguments  and they need to be kept in sync as they can go out of sync if one is updated and other one is not. 

2. Let the options be Changed before they are passed to there respective function and change the values which come from the clap::Cli enum.
Advantages :-
- It is more always make sure that every command line option can also be set with a config file and two structs will not be needed to be kept in sync of each other.
Disadvantages:-
- It is very less ridget by design and will be hard to change if a function what to keep features only by either config or by arguments

I am not used to open source much. So if the thinks are say seem too basic or are wrong please do let me know.
Also there seem to be a config feature discussion going on in the clap crate which can solve this problem entirely.

https://github.com/clap-rs/clap/discussions/2763

**Comment by tyt2y3** [comment]
Sorry for the late reply, if you want to work on it, feel free and just do whatever you think makes the most sense!

I think we always want cli params to override the config file, so it's a two pass process, we first load a config file, and then override the options with the cli params. I think the serde struct can be shared with https://crates.io/crates/structopt thus removing code duplication.

**Comment by aadi58002** [comment]
Both the clap and then serde_json during parsing the file and arguments respectively put in default value. 

```
[src/commands/generate.rs:49] parsed_config = Entity {
    compact_format: false,
    config: None,
    expanded_format: false,
    include_hidden_tables: false,
    tables: [],
    ignore_tables: [],
    max_connections: 0,
    output_dir: "",
    database_schema: "",
    database_url: "",
    with_serde: "",
    serde_skip_deserializing_primary_key: false,
    serde_skip_hidden_column: false,
    with_copy_enums: false,
    date_time_crate: Chrono,
    lib: false,
    model_extra_derives: [],
    model_extra_attributes: [],
}
[src/commands/generate.rs:50] config = Some(
    "example.json",
)
[src/commands/generate.rs:50] expanded_format = false
[src/commands/generate.rs:50] include_hidden_tables = false
[src/commands/generate.rs:50] tables = []
[src/commands/generate.rs:50] ignore_tables = [
    "seaql_migrations",
]
[src/commands/generate.rs:50] max_connections = 1
[src/commands/generate.rs:50] output_dir = "./"
[src/commands/generate.rs:50] database_schema = "public"
[src/commands/generate.rs:50] database_url = "bk"
[src/commands/generate.rs:50] with_serde = "none"
[src/commands/generate.rs:50] serde_skip_deserializing_primary_key = false
[src/commands/generate.rs:50] serde_skip_hidden_column = false
[src/commands/generate.rs:50] with_copy_enums = false
[src/commands/generate.rs:50] date_time_crate = Chrono
[src/commands/generate.rs:50] lib = false
[src/commands/generate.rs:50] model_extra_derives = []
[src/commands/generate.rs:50] model_extra_attributes = []
```

It makes it really difficult to check if some one set true in the config and then the set a variable to false from the command line it become uncertain whether the options value came from the arguments or from the default values

Any suggestion ?
code -> https://github.com/aadi58002/sea-orm/commit/87e355424b998f89ddcf81a804bf0834e4b06476

**Comment by aadi58002** [comment]
@tyt2y3 If you have suggestion for improvement let me know.
